soroban-env-host/src/events/mod.rs

```diff
@@ -123,11 +123,33 @@ impl core::fmt::Display for HostEvent {
         match &self.event.body {
             ContractEventBody::V0(ceb) => {
                 write!(f, "topics:[")?;
+
+                let mut is_fn_call = false;
                 for (i, topic) in ceb.topics.iter().enumerate() {
                     if i != 0 {
                         write!(f, ", ")?;
                     }
-                    display_scval(topic, f)?;
+
+                    // The second topic of the fn_call event is the contract id as ScBytes,
+                    // but we want to display it as a C key instead, so this block
+                    // tries to deduce if the event is the fn_call event.
+                    if i == 1 && is_fn_call {
+                        if let ScVal::Bytes(bytes) = topic {
+                            let hash: [u8; 32] = bytes.clone().0.try_into().unwrap();
```

Comment:
I'll remove this unwrap

Reply:
This should actually be handled gracefully on the off-chance if event isn't actually a real fn_call event. We should check the size first and just fallback to the default option if it's not 32.

Reply:
Yup that's what I meant by removing the unwrap. Fixed